src/Passes/docs/src/UserGuide/BuildingLibrary.md

```diff
@@ -0,0 +1,149 @@
+# Build guide
+
+This document will guide you through all the requirements and steps that you need to follow to build the QIR adaptor tool (QAT). The first section of this guide helps you install the tools needed to build QAT. After you have finished with the second section of the guide, you should have the `qat` executable compiled and ready to run. As a part of the third part of this post, we will show you how to build and serve all of the documentation including the user guides, developer guides and API documentation. We will go through the steps of compiling the whole library as well as how to run all the tests suite that has been provided as the fourth part of this document. This part is only relevant if you intend to extend or modify the core codebase. There is no need for it when simply building an extension to QAT without having the need to change anything else in the core library. There is a lot more information about how to create a custom extension within the developers’ section.
+
+## Prerequisites
+
+It is necessary for you to have the following tools before you get started on making a new build from this project:
+
+- C++ compiler
+- CMake
+
+If you are compiling and using the command-line tool, these are necessary tools regardless of whether you plan to develop the library itself. The build configuration is handled by CMake, which informs you if a compiler needs to be installed.
+
+### On OS X
+
+Those using OS X can install CMake through the use of a command-line tool called `brew`, as follows:
+
+```sh
+brew install cmake
+```
+
+It should not even be necessary to install anything else on top of OS X, since the compiler comes preinstalled. There is no need to read the next subsection ([Developer prerequisites](#developer-prerequisites)) if you are not intending to modify the core library and can safely jump to the section [Library Dependencies](#library-dependencies)
```

Comment:
```suggestion
It should not even be necessary to install anything else on top of macOS, since the compiler comes preinstalled. There is no need to read the next subsection ([Developer prerequisites](#developer-prerequisites)) if you are not intending to modify the core library and can safely jump to the section [Library Dependencies](#library-dependencies)
```

Reply:
Updated.